File: ryan/week3/medium.py
def solution(cards1, cards2, goal):
    cards1.append("")
    cards2.append("")
    p_1 = 0
    p_2 = 0
    for i in goal:
        if i == cards1[p_1]:
            p_1 += 1
        elif i == cards2[p_2]:
            p_2 += 1
        else:
            return "No"
    return "Yes"

File: dongwoo/week3/medium.java
class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        String answer = "Yes";
        int count = 0;
        int glen = goal.length;
        int p1 = 0;
        int p2 = 0;
        while (count < glen) {
            String g = goal[count];
            if (p1 < cards1.length) {
                if (g.equals(cards1[p1])) {
                    p1++;
                    count++;
                    continue;
                }
            }
            if (p2 < cards2.length) {
                if (g.equals(cards2[p2])) {
                    p2++;
                    count++;
                    continue;
                }
            }

            return "No";

        }
        return answer;
    }
}
